apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx

```diff
@@ -316,22 +317,19 @@ class VisualizerModal extends React.Component {
               }
               inlineLabel
             />
-            <DropdownField
-              displayName={msg.dataVisualizerCreateChart()}
-              options={[]}
-              disabledOptions={[]}
-              value={this.state.screen}
-              onChange={event => this.setState({screen: event.target.value})}
-              inlineLabel
-            />
-            <button
-              type="button"
-              style={dataStyles.grayButton}
-              onClick={this.handleOpen}
-            >
-              {msg.create()}
-            </button>
           </div>
+          <Snapshot
+            records={filteredRecords}
+            numericColumns={numericColumns}
+            chartType={this.state.chartType}
+            bucketSize={this.state.bucketSize}
+            chartTitle={this.state.chartTitle}
+            selectedColumn1={this.state.selectedColumn1}
+            selectedColumn2={this.state.selectedColumn2}
```

Comment:
Looks like most of these props are passed directly to `DataVisualizer` through `Snapshot`. Consider factoring your `DataVisualizer` component on line 276 into a display function. Then pass it as a child component here.

i.e. 
```
renderDataVisualizer = (fiteredRecords, numericColumns) => {
  return (
    <DataVisualizer
      records={filteredRecords}
      numericColumns={numericColumns}
      chartType={this.state.chartType}
      bucketSize={this.state.bucketSize}
      chartTitle={this.state.chartTitle}
      selectedColumn1={this.state.selectedColumn1}
      selectedColumn2={this.state.selectedColumn2}
    />
  );
};

...

<Snapshot
  filterColumn={this.state.filterColumn}
  filterValue={this.state.filterValue}
  isMultiColumnChart={isMultiColumnChart}
>
  {this.renderDataVisualizer}
</Snapshot>
```